olive/engine/engine.py

```diff
@@ -816,6 +816,10 @@ def _run_pass(
         pass_config = p.config_at_search_point(pass_search_point)
         pass_config = p.serialize_config(pass_config)
 
+        # set evaluator to pass
```

Comment:
the evaluator info will be dropped when running in AzureML case. 

@jambayk , do we need support it?

Reply:
Yes, I think we will need to support it. However, I am not sure what the best way to do this is. 

I think it might be better to insert the evaluator config into the pass config like we do for `data_config` instead of setting a model attribute to the pass object. What do you think @trajepl?

Reply:
Agree with Jambay‘s idea. Something like the flag of `_requires_user_script` in Pass's attribute, then we can replace the evaluator_config for specific pass.

But think it a bit more[rough idea], we have many components like `user_script/script_dir`, `data_config`, `evaluator`, `aml_client` and etc. are required by engine/pass/system etc.  Maybe we should propose the component like central component in olive?
It may not change the user-facing config. But will give a clear definition and unified way for engine/pass/system to pull the corresponding configs from this central component.

Reply:
does this mean we need allow pass refer to the evaluation config defined in "evaluator" section? or we need copy the config and put them as plain text in pass config?

Reply:
Currently, the latter one which copy the config. Because Olive need to ensure pass run can be directly started even there is no outside evaluation config.

Reply:
I suppose reference to evaluator defined in evaluators section should be able to work and it is more concise than the latter solution. Are there any blockers?

Reply:
If you mean the reference like olive system, I think we are at the same line. I am just thinking you may mean the reference to the config object.

Finally, the run_config.parser will replace the reference automatically with the full config.

Reply:
Yeah, I mean the behavior like system target or host.

Reply:
@yuwenzho , please ping @trajepl if you need help on how to do the change.

Reply:
sure, we will do that after the Chinese Dragon Boat Festival. thanks for your support.